### PR TITLE
[4.0] Rename `T` to `Value` and `E` to `Error`

### PR DESCRIPTION
**Pull request by bencochran**
This mostly completes #2212. I left the `V` and `F` parameters alone… it’s what the standard library mostly does, and things like `TransformedValue` make for some long function signatures. But I could be convinced otherwise.

`Event` presented a challenge: it already has a static member named `Error` (namely, the `case Error(E)`). So it cannot be generic with `Error` as the parameter name. I left it as `E` for now. Anyone have ideas for a name?

### CODE REVIEW

File: ReactiveCocoa/Swift/Signal.swift [hunk 1]
```diff
@@ -722,9 +722,9 @@ extension SignalType {
 	/// Waits until `self` completes and then forwards the final `count` values
 	/// on the returned signal.
 	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
-	public func takeLast(count: Int) -> Signal<T, E> {
+	public func takeLast(count: Int) -> Signal<Value, Error> {
 		return Signal { observer in
-			var buffer = [T]()
+			var buffer: [Value] = []
```
Comment (bencochran) [inline ReactiveCocoa/Swift/Signal.swift]: I snuck in this change to be more stylistically self-consistent.

**Comment by neilpa** [comment]
The only reasonable name I've got is `Err`, but abbreviating like that isn't idiomatic.

Alternatively, the `Error` case could be renamed `Failure`. It would mirror the naming in `Result`, but I would need more convincing about making this change.

**Comment by bencochran** [comment]
Hmm, interesting thought about changing the case name. I could also see `Failed` which mirrors `Completed`/`Interrupted` in its tense (and since those are all terminating events, having the same sense is nice).

**Comment by neilpa** [comment]
:+1: on `Failed` over `Failure` if we go down that route.

**Comment by bencochran** [comment]
Ok, I rebased this to pull in @NachoSoto’s stuff. And I used `Err` in `Event` for now.

I’m going to do a branch with `case Failed` and see how that feels, but I’ll let that be a separate pull request and discussion.

**Comment by neilpa** [comment]
> but I’ll let that be a separate pull request and discussion

:+1:

**Comment by bencochran** [comment]
Opened that thing as a separate pull request targeting the swift2 branch. So this should be good to go on its own.

**Pull request closed without merging**